### DISCUSSION TITLE
Object burn at a specified rate

### DISCUSSION THREAD

**Discussion by RNstu08** [firemodels/fds#12845] [Q&A]
Hello Everyone,

I am modelling a fire curve from the experiment using HRRPUA model. Basically, I have a HRR curve from the experiment and I used this as a input on the surface to release heat at a specified rate. Additionally, I assigned the ignition temperature to the surface (i,e,. it should starts to burn once it reaches this temperature following a ramp). I used burner at the bottom with 100kW for 10min. As a test run, I am using PVC material properties from the FDS library and Heptane was used for the gas phase reaction. It was run for 1200 sec.
My aim is to replicate the HRR curve found in the experiment.

In this case. when I used ignition temperature on the surface the material, the material burns but the HRR obtained is different from the input curve. However, when I remove the ignition temperature on the surface and assumes that it should burn immediately, I obtain the HRR curve similar to experiment.

My question is, after reaching ignition temperature, why isn't it following the input HRRPUA curve?
When I prescribed the HRR, I should be able to obtain the same HRR curve, right?
Any inputs would be deeply appreciated.

Thank you.
[Test_run.txt](https://github.com/firemodels/fds/files/15097703/Test_run.txt)

**Comment by drjfloyd** [comment]
When you use IGNITION_TEMPERATURE, it is evaluated on a cell-by-cell basis. Each cell will start follwing the HRRPUA curve when it reaches the ignition temperature and not all cells at once.

**Reply by RNstu08** [reply]
But, in these type of simulation where the HRRPUA is prescribed, the simulation should replicate the HRR curve irrespective of the case, right?

**Reply by drjfloyd** [reply]
When you use IGNITION_TEMPERATURE, then HRRPUA gets applied to each grid cell when it ignites.  Say you have two grid cells and your HRRPUA is 1000 kW/m2 for 10 s, your grid cells are 0.1 m2, and both ingnite at 0 s.  What will the HRR over time in FDS be?

Now one griid cell ignites at 0 s and one ignites at 10 s.  Now what is the HRR over time?

**Reply by RNstu08** [reply]
For example, if I use ignition_temperature of 80C. Then 1000kw/m2 HRRPUA used as a input applied each grid cell when it reaches the ig_temp. Area of two grid cells are 0.1m2. one grid area is 0.05m2.

If it ignites at 0sec, that no ignition _temperature is assigned, assuming it burns immediately, HRR will be 100kW over time

If one grid ignites immediately at 0s, then for this grid will emit 50kW from start over time.  And another grid starts to emit 50kW after 10s over time.

Is it right?

**Reply by drjfloyd** [reply]
The HRRPUA profile was 1000 kW/m2 for 10 s. So case 1 would have a fire that lasts 10 s that is the HRRPUA applied to the area of both grid cells. Case 2 would have the first grid cell ignite and burnout when the second grid cell starts. So case 2 would have a 20 s fire that is the HRRPUA applied to one grid cell.

Now think about what this means for your original question where in one case you have the entire surface ignite at t=0 and in the other case each grid cell ignites when it reaches its ignition temperature. Would you expect the same HRR?

**Reply by RNstu08** [reply]
In case 1, HRR obtained from the start.
In case 2, HRR obtained when it reaches the ignition temperature criteria. That means, in case 2, HRR obtained is half of that obtained in case 1 from the start.

That means if ignition temperature criteria is used, it is not possible to replicate the prescribed HRR curve. Right?

**Reply by drjfloyd** [reply]
The curve you prescribe in the input is not an HRR curve; it is an HRRPUA curve. Each well cell replicates the prescribed HRRPUA curve. It is unlikely with IGNITION_TEMPERATURE that all wall cells will ignite at the same time so the HRR is unlikely to equal the surface area times HRRPUA as it does when you do not use IGNITION_TEMPERATURE.

**Reply by RNstu08** [reply]
That means, in order to replicate the input HRRPUA (Fire curve of the experiment), one has to use the assumption of burn immediately.
If I use Ignition temperature criteria, it is not possible to replicate the curve.

**Comment by mcgratta** [comment]
When you use the `IGNITION_TEMPERATURE` approach, you are, in some sense, PREDICTING, the HRR. When you simply specify `HRRPUA` over a constant area, you are SPECIFYING the HRR. These are two different approaches. If you want a specific HRR curve, do not use the `IGNITION_TEMPERATURE` approach.

**Reply by RNstu08** [reply]
Thank you, I understand. How do I use this for validation of experiment?

For example, if I conduct an experiment. Where I burned a object using 10kW burner placing below it and I also found ignition temperature of this object. I obtain some HRR as a result of this.
If I want to validate this case in FDS, I will have to specify ignition temperature, HRRPUA for the object and place the burner. However, the HRR result from FDS does not resemble.
How to validate the case? Am I thinking wrong?

**Comment by mcgratta** [comment]
In the latest version of FDS, there is a feature called "Scaling Pyrolysis" or S-pyro. Take a look at that. You are going to need experimental data from a cone calorimeter or similar device to give you an HRRPUA curve plus an ignition temperature.

**Reply by RNstu08** [reply]
But I am trying to understand for the experiments that are not conducted in cone calorimeter.

For instance, I just have a Fire curve from experiments (maybe from some reference paper) not from calorimeter. For this, I performed three simulations and validated using experimental curve.
In Sim1: I used burner under the object and with no ignition temperature specified for the object surface. FDS results matched with experiments.
In Sim2: I used burner under the object and an ignition temperature was specified for the object surface. FDS results did not matched with experiments.
In Sim3: I did not use the burner rather I used the FDS example case (specified_hrr.fds). Where object surface give an external flux value and also an ignition temperature was specified. FDS results matched with experiments.

Why the FDS results from Sim2 did not follow the HRRPUA curve which was given as a input? I understand that all the grid cells in this case did not burn immediately once the object reaches the ignition temperature because all the grid cell did not reach the ignition temperature at once. That means one could not be able to use this at all? What if I know the HRR curve and ignition temperature from experiments, and I simply want to convert HRR to HRRPUA to the constant surface area of the object and assign ignition temperature to it and and validate the FDS results with experiment? Isn't it realistic? 


![Graph1](https://github.com/firemodels/fds/assets/97069157/3e5474bd-ce31-4749-a3e8-2a8a35f2474f)
[Sim1.txt](https://github.com/firemodels/fds/files/15119956/Sim1.txt)
[Sim2.txt](https://github.com/firemodels/fds/files/15119957/Sim2.txt)
[Sim3.txt](https://github.com/firemodels/fds/files/15119958/Sim3.txt)

**Comment by drjfloyd** [comment]
HRR is not the same as HRRPUA; one is kW and the other is kW/m2. 

For all three cases FDS is following the HRRPUA that you input.
-Case 1 the entire surface starts burning right away. You obtained HRRPUA by dividing HRR by the area of object, so when all cells start following the HRRPUA ramp at the same time, you simply get back the HRR curve.
-Case 3 you have IGNITION_TEMPERATURE and EXTERNAL_FLUX. That EXTERNAL_FLUX is applied to all cells of the surface heating them at nearly the same rate (there will be some minor differences from convection). As a result the cells will ignite at nearly the same time and in essence you just have Case 1 with delayed igntion.
-Case 2 when each cell reaches the IGNITION_TEMPERATURE it starts following the HRRPUA ramp you input. Because the cells do not ignite at the same time (see the simple example I gave previously), you will not get back the HRR curve. But each wall cell is using the HRRPUA that you input.

If you just want to follow an HRR curve, do not use IGNITION_TEMPERATURE. You can use devices and/or control functions to start the HRRPUA RAMP based on the surface temperature. See the User's Guide for details.

**Reply by RNstu08** [reply]
It would be great if you could give pointer in the user guide. Thank you

**Reply by drjfloyd** [reply]
Search the table of contents for a chapter that discusses using devices and control logic.

**Comment by RNstu08** [comment]
I think this how it should look like.
&VENT ID='Side_vent', SURF_ID='Surface02', XB=2.0,2.0,0.0,1.5,0.1,0.5, DEVC_ID=Wall_Temperature_MAX/ 
&DEVC ID='Wall_Temperature_MAX', QUANTITY='WALL TEMPERATURE', SPATIAL_STATISTIC='MAX', XB=1.9,2.1,0.0,1.5,0.1,0.5, SETPOINT=80/
However, this is indirectly using the case where we don't assign ignition temperature rather use setpoint and all the cell grid burns once the vent reaches the set point.

**Reply by drjfloyd** [reply]
Try it and see if it does what you want. That is the best way to learn how to use FDS.

**Reply by RNstu08** [reply]
I tried it, the results are same when I don't use ignition temperature (i.e., Sim1.txt).

But, in reality, when the experiment is conducted, the solid object initially gets heated upto certain temperature and when it reaches its critical ignition temperature the object starts to burn, right? That's how the chosen design curve is interpreted, right?
If so, it should similar to my Sim2.txt simulation. Then, why is the HRR results from this simulation is different than the experiments?
Would you please explain it?

**Reply by drjfloyd** [reply]
When you burn a real world object, yes some spot on the object reaches some kind of ingition threshold, starts burning, and fire then spreads over the object.  If you are measuring the heat release rate of the entire object, you don't know the HRRPUA(x,y,z,t) over the entire surface of the object. All know from a calorimeter is HRR(t). You can get an HRRPUA(t) by dividing by the total burning area, but that is not the same as the HRRPUA(x,y,z,t).  When you use HRRPUA(t) and IGNTION_TEMPERATURE, you are making the assumption that HRRPUA(x,y,z,t) = HRRPUA(t). It appears for this object, for the HRR data you have, that this is not a good assumption.

**Reply by RNstu08** [reply]
Could you please explain what are the other possibilities or approaches or assumptions that are possible?

**Reply by drjfloyd** [reply]
Perform experiments to develop detailed kinetics. Search the literature for how to do this.
Perform experiments to understand the burning behavior of the components of the object at a non-integral length scale (like a cone test) to try and get a better estimate of HRRPUA(x,y,z) and ignition temperature. 
Do nothing and either just apply the HRR curve to the entire object at once or live with the prediction you are getting with ignition temperature.

**Reply by RNstu08** [reply]
Okay. I have follow up question when we have two objects next to each other.

Suppose, First object is burning as in sim1.txt (i.e., without ignition temperature and specifying HRRPUA) and second object is placed some distance (for ex 30cm in x direction). Second object is also specified with same HRRPUA and without ignition temperature. I got total heat release rate from the FDS which is bigger than the specified curve. This is because two objects are burning and second object starts to burn at once when one of it's grid cells ignited.
My question is, how to ensure that second object is burning with specified HRRPUA? Is it by using burning rate DEVC option around the second object?
How do I know when is it (second object) ignited or at which point it (second object)  is ignited? As I am not giving ignition temperature criteria to it.

**Reply by rmcdermo** [reply]
I would just add a BNDF of QUANTITY='BURNING RATE', then check the total HRR matches what you expect.

**Reply by RNstu08** [reply]
I am further analyzing the behavior observed in Sim2.

In this case, I have a burner under the first object (Object1) with a prescribed HRRPUA. For the second object (Object2), I have added an ignition temperature along with the HRRPUA for the object's surface.

I want the first object to burn immediately and follow the HRRPUA curve. The second object should ignite at the specified ignition temperature and then follow the HRRPUA curve as well.

FDS predicted the HRR from the first object very well (i.e., it matched the input curve as expected). However, the burning of the second object results in a repeated HRR profile. This repeated curve follows my input HRRPUA curve but repeats over and over. I have attached the HRR graph here along with the .fds file.

I would greatly appreciate it if someone could provide insights into the behavior observed with the burning of the second object. Shouldn't it stop burning according to the input HRRPUA profile? How do I control this repeated burning behaviour?

I have tried several different cases, such as modifying material properties, using different reaction properties, and trying the recommended method using DEVC. However, in all cases, the second object exhibits the same repeated behavior.

![Graph1](https://github.com/user-attachments/assets/c3f24cfa-43bc-40fc-b656-5c2f5f86222f)
[Objects.txt](https://github.com/user-attachments/files/16573094/Objects.txt)

Thank you all in advance.

**Reply by drjfloyd** [reply]
As stated previously on April 26, SURF with IGNITION_TEMPERATURE operates cell-by-cell and not obst-by-obst.

What you are plotting is not HRR.  Your device is BURNING RATE with SPATIAL STATISTIC of MAX. This is not the total burning rate of the OBST and it is defintely not the realized HRR in the domain as the domain is too small to allow that HRR to be realized.. You are only outputting the maximum burning rate at any point within the XB.

You are using a CFD trying to understand the behavior of burning rate with an object with an igntion temperature but your only outputs other than the default hrr.csv file is the max burning rate. Why not add BNDF for BURNING RATE or WALL TEMPERATURE so you could have seen exactly what is happening for the two OBST?

**Reply by RNstu08** [reply]
I understand that the simulation operates on a cell-by-cell approach when using ignition temperature as a surface boundary condition. However, my goal is to see whether the second object follows the input HRRPUA once it is ignited, specifically to observe if it follows the prescribed HRRPUA profile and stops burning once the HRRPUA is fully consumed or finished. Is it possible to achieve this within the current simulation framework?

The burning rate I used here is to determine the HRR from both objects. I calculated the HRR by taking the burning rate across the surfaces of the objects, multiplying it by the heat of combustion, and considering the surface area. This approach allowed me to generate the HRR graphs that I shared previously. I also understand that the domain is small, and therefore, the obtained default HRR.csv does not fully realize the expected HRR.

I did use BNDF quantities earlier, which provided a visualization of the objects and what is happening at the surfaces. However, I excluded these results because I could not directly compare them with my input HRRPUA profile.

What I don't understand is why the first object follows the input HRRPUA profile accurately, reflecting the same HRR curve when I calculate it using the burning rate as described earlier, while the second object repeats the input HRRPUA curve in its output instead of consuming the fuel fully. Why is the second object behaving like it's in a repeated loop, rather than consuming the fuel as expected?

**Comment by Zaninho23** [comment]
hello everyone, I write in this discussion because I tried to implement the simulation described by 
@RNstu08. In particular, I created three simulations, in the first one I inserted a simple 1x1x1m cubic burner characterized by the following HRRPUA curve:
![image](https://github.com/firemodels/fds/assets/38040558/a9df56f7-73ec-45de-b1a3-b5c18d814d76)

Running the analysis, the HRR curve obtained is as follows:
![image](https://github.com/firemodels/fds/assets/38040558/86315c4c-31c3-4663-986b-620a19f4d6f7)
It corresponds, correctly, to the HRRPUA scaled for the fuel surface area of 5m2. 

In the second simulation I copied such a burner and, as expected, the HRR curve is the result of the sum of the two curves point by point:
![image](https://github.com/firemodels/fds/assets/38040558/2bba7b51-e282-4b5e-b879-eb96ad889185)

In the third simulation instead of one of the two burners I inserted a cube of "wood" characterized by an HRRPUA always the same, but a temperature ignition equal to 150°C. In the latter case as also evidenced by the tests done by @RNstu08 and in agreement with the behavior described by @drjfloyd (definable as cell to cell) a different HRR curve is obtained. What I do not understand, however, is why even though the second cube participates in the combustion it does not "burn" completely. Even from the smokeview images it appears that only some faces (the top one and the one toward the burner) participate in the combustion with the remaining portion of the cube not being involved. 
![image](https://github.com/firemodels/fds/assets/38040558/0638148e-ee23-4005-801d-b155d2db1f4c)
![image](https://github.com/firemodels/fds/assets/38040558/7fe067ce-13d7-4464-a5f5-f5cb9780abf4)

What could be due to such behavior ? to the thermal inertia of the material I have inserted ?
Thank you very much.
[SIM_1.txt](https://github.com/user-attachments/files/15864582/SIM_1.txt)
[SIM_2.txt](https://github.com/user-attachments/files/15864583/SIM_2.txt)
[SIM_3.txt](https://github.com/user-attachments/files/15864584/SIM_3.txt)

**Comment by drjfloyd** [comment]
You can ouput pretty much every quantity that FDS predicts. As an FDS user you should have sufficient outputs to allow you to understand what is happening in your simulation. In your third case, wall cells will only ignite and start burning if they reach 150 C.  You have no outputs showing the wall temperatures. First thing to do is to see what temperatures the target cube is reaching. If wall cells don't reach 150 C, then they won'y burn.

**Reply by Zaninho23** [reply]
Of course, the surfaces do not reach 150°C. From the simulation it seems that the heat and therefore the flame does not go beyond the corners of the cube and downwards. 
![image](https://github.com/firemodels/fds/assets/38040558/1329fa89-c190-469e-b703-56fec8d4be6c)

In your experience, is this behavior due to combustion modeling (simple chemistry) without pyrolysis and cell to cell ignition?

**Reply by drjfloyd** [reply]
The heat transfer in your input file is 1D; i.e. there is no wall cell-to-wall cell heat trasnfer. The sides have no view factor of the fire to be able to heat up and burn.

**Reply by Zaninho23** [reply]
Thank you very much, now I understand the problem. I will look up the 3D heat transfer in the manual to see how it works. In other applications I can instead use a controller that turns on the whole object at a certain temperature, which I think is the easiest thing to do.

Thanks again for the help.

**Reply by drjfloyd** [reply]
Add a BNDF for BURNING RATE andd WALL TEMPERATURE and look at the output in Smokeview.  You plotted the MAXIMUM burning rate over all wall cells in the XB not the burning rate of the entire obstacle. Do all the wall cells in the second OBST ignite at the same time? If not, what implications does that have for maximum value at any instant in time over the second OBST.

**Reply by RNstu08** [reply]
I have removed BNDF as suggested for this discussion. However, with this I can only visually see how the burning rate and wall temp evolve over time on the surface of the obstructions.

I had plotted max burning rate over all wall cells because by multiplying (burning rate * hc * surface area of the obstruction), with this i get to compare the fds output with the input curve. I understand that object1 or object2 are composed of many cells and each cell on the surface may ignite at different times depending on the local conditions. At any given instant, the max value captured does not represent the total burning rate of the object rather the most intense burning cell that moment. With maximum burning rate DEVC, I am capturing the highest burning rate among all the surface cells in the defined XB at any given time. Isn't it how it supposed to be if I want to compare the fds output with input curve?

Because, if I take the burning rate integral, which would sum the burning rate over all the surface cells within the XB of the obstacle. This would give me the overall rate of fuel consumption by the entire surface of the obstacle. This comprehensive view of how much of the obstacle burning over time for the 'object2' failed when compared with input curve.


By using ignition temp on the second object, I understand that all the wall cells in the second OBST wont ignite all the same time rather depending on the conditions at the cell. This suggest, tht at any instant of time, max value captured does not represnt the total burning rtae of the object rather the most intense burning cell.
![image](https://github.com/user-attachments/assets/4758322d-fc2c-477f-9be2-2f7aa349c0a6)

But my question still remains the same, why does second object burning like repetitive loop similar to given input profile?

**Comment by drjfloyd** [comment]
Take this RAMP:
T=0,F=0
T=10,F=1
T=20,F=1
T=30,F=0

You have an obst the ignites cell-by-cell with IGNITION_TEMPERATURE (as I have told oyu twice now IGNITION_TEMPERATURE works cell-by-cell). One cell ignites at 5 s, once cell ignites at 10 s, and one cell ignites at 35 s.  What would the ouptut of your MAX device look like just using the values of the RAMP?  Make a plot for each wall cell and then a plot for the device.

**Reply by RNstu08** [reply]
![Screenshot_2024-08-12-12-46-23-31_92460851df6f172a4592fca41cc2d2e6](https://github.com/user-attachments/assets/8d66de1b-a6d8-4a91-82f8-8029c24d8e54)
![IMG_20240812_124722](https://github.com/user-attachments/assets/508d2fb2-c3f6-485c-a250-e2c67422fe8f)

Cell 1 and Cell 2 burn at full rate between 10s and 20s, causing the MAX device with a peak value of 1 during that time. After 30s, the burning rate drops to zero as no cells are burning. Cell 3 never burns because it ignites after the RAMP has already decreased to zero.

In the MAX device plot initially shows zero until Cell 1 ignites at 5 seconds.
And, from 10 to 20 seconds, it shows the peak burning rate because both Cell 1 and Cell 2 are burning at their full rate. After 20 sec, as the RAMP decreases, the burning rate reduces until it reaches zero at 30 sec. After 30 sec, the MAX device shows zero since no cells are burning.

In my previous graph, when I compared using MAX devc, the peak hrr obtained (after calculating as mentioned earlier) from both devices for both the objects were 6000kW which is same as that of my input graph's peak. Whereas, in the DEVC with surface integral statistics, the peak obtained for the second object was less than 3000kw. If I compare this with my input then obviously it is just half of the input curve. Why is there such a difference? 

The other question, is the repeated behavior observed in the MAX Devc curve related to energy content of the object?  Does it mean there is more fuel content and hence it keeps on burning?  But in that I used below ramp for second object, why it is not stop burning once the ramp is finished (at 240sec) for the second object?
T=0.0, F=0.0/
T=60.0, F=1.0/
T=180.0, F=1.0/
T=240.0, F=0.0/
Thanks you again

**Reply by drjfloyd** [reply]
Your plots are not correct.  The first cell ignites at 5 s. It will be zero until  5 s, and then follow the ramp. The second starts its ramp at 10 s and the last at 35 s.  Try again.

**Reply by RNstu08** [reply]
![image](https://github.com/user-attachments/assets/e902407c-dc58-41ce-92b3-cc4c7a913ce2)
![image](https://github.com/user-attachments/assets/809f6a1e-b36c-4226-81e1-40b36ba2cf70)
That means the second object has multiple cells and each cell follow the ramp and therefore with the MAX devc, it looks like repeated curve. Right?
What would be the better way to approach when I use the ignition temperature approach? How do I know that the second object is following the ramp? It looks like my simulations are following. But, what I dont understand is the way to justify. Because, the input ramp is for entire second obstruction and hence measurement should be made from the entire object.

By measuring the burning rate with MAX statistics it showed same peak as that of the input ramp. Whereas, with the burning rate integral statistics, understandably the ramp has decreased when compared with the peak in both input and fds output.
Does this mean, with the ignition temperature surface boundary condition, one can never get the same input profile like one get when all the cells are assumed to burn immediately?
Thank you

**Reply by drjfloyd** [reply]
Yes exactly. The first curve you posted shows a slightly longer plateau and a second peak. This is what you would expect if in your simulation most cells one the second obst ignited close to one another (plateaus overlap giving a longer plateau when measured by MAX) but some cell or cell(s) for whatever reason ignites later (giving a second rise to a plateau when measured by MAX).

If you want the entire second OBST to simply burn at a prescribed rate over the entire OBST whenever any cell reaches the ignition temperature, then put VENTs on the six surfaces and activate the VENTs whenever some wall cell reaches the igntion temperature.
